### PR TITLE
Add 1x and 2x avatar image variant

### LINKED ISSUE #519: Serve avatars at 1x and 2x
As of #518, we generate variants for avatars @2x the displayed size instead of scaling down the full image. For instance, avatar thumbnails are displayed at 32x32, while the thumbnail itself is 64x64.

As per [this comment](https://github.com/joemasilotti/railsdevs.com/pull/518#issuecomment-1164461961), we want to push this further this by also serving avatars at 1x and 2x via HTML.

## A PR addressing this issue should:

- [ ] Create HTML markup that ensures 1x and 2x images are rendered for the correct screen resolution

I _think_ this might involve using the `srcset` attribute on `` but can't say for sure.

### PR DESCRIPTION
**Pull request by adrienpoly**
close #519

This PR enhance the AvatarComponent to provide both 1x and 2x images. This is a performance improvement, standard monitor will receive the 1x image (32x32) for thumbnail. Whereas Retina monitors will receive the 64x64. Everything is handled by the standard `srcset` html attribute.

I added 2 new variants in the `Avatarable` concern `thumb_2x` and `medium_2x`. From my test locale with a db having only the previous variant, Rails generate automatically the new ones on the first request.

## Pull request checklist

- [x] My code contains tests covering the code I modified
- [x] I linted and tested the project with `bin/check`
- [ ] I added significant changes and product updates to the [changelog](CHANGELOG.md)

### CODE REVIEW

File: app/models/concerns/avatarable.rb [hunk 1]
```diff
@@ -3,8 +3,10 @@ module Avatarable
 
   included do
     has_one_attached :avatar do |attachable|
-      attachable.variant :thumb, resize_to_limit: [64, 64]
-      attachable.variant :medium, resize_to_limit: [256, 256]
+      attachable.variant :thumb, resize_to_limit: [32, 32]
```
Comment (joemasilotti) [inline app/models/concerns/avatarable.rb:6]: Is there any risk in the variant being named the same thing but the size changed... so we would never see the correct variant size?

Comment (adrienpoly) [reply app/models/concerns/avatarable.rb:6]: from my test using a seeded data from #530 and then adding those variant it is working as expected but a double check could be useful indeed

File: test/components/avatar_component_test.rb [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 class AvatarComponentTest < ViewComponent::TestCase
   include ActionView::Helpers::AssetUrlHelper
+  include ActionView::Helpers::UrlHelper
+  include Rails.application.routes.url_helpers
```
Comment (adrienpoly) [inline test/components/avatar_component_test.rb:6]: needed those lines ignorer to use `url_for` not sure if someone has a better approach for that

File: app/models/concerns/avatarable.rb (hunk 1 above)
Comment (joemasilotti) [reply app/models/concerns/avatarable.rb:6]: Nice. I'm going to run the backfill again on deploy, just in case.

**Comment by joemasilotti** [comment]
Awesome – thanks for this! Going to merge in now.

**Pull request merged**